From: Michael Froman <mfroman@mozilla.com>
Date: Thu, 10 Oct 2024 13:42:00 +0000
Subject: Bug 1921707 - point to new build directory location in
 third_party/abseil-cpp r=ng

Ran:
ack -l "\"//build/config" third_party/abseil-cpp | xargs sed -i.bak -E 's/\"\/\/build\/config/\"\/\/chromium\/build\/config/'
ack -l "\"//build/toolchain" third_party/abseil-cpp | xargs sed -i.bak -E 's/\"\/\/build\/toolchain/\"\/\/chromium\/build\/toolchain/'
find third_party -name "*.bak" | xargs rm

Differential Revision: https://phabricator.services.mozilla.com/D224544
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/566543823f7e89dc33d6d59a438aea7a1d19e3b0
---
 abseil-cpp/.gn                     |  2 +-
 abseil-cpp/BUILD.gn                |  8 ++++----
 abseil-cpp/absl.gni                | 14 +++++++-------
 abseil-cpp/absl/base/BUILD.gn      |  4 ++--
 abseil-cpp/absl/debugging/BUILD.gn |  2 +-
 5 files changed, 15 insertions(+), 15 deletions(-)

diff --git a/abseil-cpp/.gn b/abseil-cpp/.gn
index 648423a128f..bd6c38b3762 100644
--- a/abseil-cpp/.gn
+++ b/abseil-cpp/.gn
@@ -1,5 +1,5 @@
 # The location of the build configuration file.
-buildconfig = "//build/config/BUILDCONFIG.gn"
+buildconfig = "//chromium/build/config/BUILDCONFIG.gn"
 script_executable = "python3"
 secondary_source = "//abseil-cpp/"
 
diff --git a/abseil-cpp/BUILD.gn b/abseil-cpp/BUILD.gn
index ea54e47e69f..e4e95660075 100644
--- a/abseil-cpp/BUILD.gn
+++ b/abseil-cpp/BUILD.gn
@@ -6,10 +6,10 @@
 # opts will be linked together, and in some cases headers compiled with and
 # without these options will be part of the same program.
 
-import("//build/config/c++/c++.gni")
-import("//build/config/nacl/config.gni")
-import("//build/config/sanitizers/sanitizers.gni")
-import("//build/toolchain/toolchain.gni")
+import("//chromium/build/config/c++/c++.gni")
+import("//chromium/build/config/nacl/config.gni")
+import("//chromium/build/config/sanitizers/sanitizers.gni")
+import("//chromium/build/toolchain/toolchain.gni")
 import("//build_overrides/build.gni")
 import("//third_party/abseil-cpp/absl.gni")
 
diff --git a/abseil-cpp/absl.gni b/abseil-cpp/absl.gni
index 9de3d1c9653..ed50a0f6e65 100644
--- a/abseil-cpp/absl.gni
+++ b/abseil-cpp/absl.gni
@@ -30,10 +30,10 @@ template("absl_source_set") {
       not_needed(invoker, "*")
     } else {
       forward_variables_from(invoker, "*")
-      configs -= [ "//build/config/compiler:chromium_code" ]
+      configs -= [ "//chromium/build/config/compiler:chromium_code" ]
       configs += [
-        "//build/config/compiler:no_chromium_code",
-        "//build/config/compiler:prevent_unsafe_narrowing",
+        "//chromium/build/config/compiler:no_chromium_code",
+        "//chromium/build/config/compiler:prevent_unsafe_narrowing",
         "//abseil-cpp:absl_default_cflags_cc",
         "//abseil-cpp:absl_define_config",
       ]
@@ -41,8 +41,8 @@ template("absl_source_set") {
       if (is_component_build) {
         defines = [ "ABSL_BUILD_DLL" ]
         if (!is_win && current_os != "aix") {
-          configs -= [ "//build/config/gcc:symbol_visibility_hidden" ]
-          configs += [ "//build/config/gcc:symbol_visibility_default" ]
+          configs -= [ "//chromium/build/config/gcc:symbol_visibility_hidden" ]
+          configs += [ "//chromium/build/config/gcc:symbol_visibility_default" ]
         }
       }
 
@@ -143,9 +143,9 @@ template("absl_test") {
     } else {
       forward_variables_from(invoker, "*")
       testonly = true
-      configs -= [ "//build/config/compiler:chromium_code" ]
+      configs -= [ "//chromium/build/config/compiler:chromium_code" ]
       configs += [
-        "//build/config/compiler:no_chromium_code",
+        "//chromium/build/config/compiler:no_chromium_code",
         "//abseil-cpp:absl_default_cflags_cc",
         "//abseil-cpp:absl_define_config",
         "//abseil-cpp:absl_test_config",
diff --git a/abseil-cpp/absl/base/BUILD.gn b/abseil-cpp/absl/base/BUILD.gn
index de99dd146b0..6a76b401fc1 100644
--- a/abseil-cpp/absl/base/BUILD.gn
+++ b/abseil-cpp/absl/base/BUILD.gn
@@ -235,9 +235,9 @@ absl_source_set("pretty_function") {
 # just remove it.
 # source_set("exception_safety_testing") {
 #   testonly = true
-#   configs -= [ "//build/config/compiler:chromium_code" ]
+#   configs -= [ "//chromium/build/config/compiler:chromium_code" ]
 #   configs += [
-#     "//build/config/compiler:no_chromium_code",
+#     "//chromium/build/config/compiler:no_chromium_code",
 #     "//third_party/abseil-cpp:absl_test_cflags_cc",
 #   ]
 #   public_configs = [ "//third_party/abseil-cpp:absl_include_config" ]
diff --git a/abseil-cpp/absl/debugging/BUILD.gn b/abseil-cpp/absl/debugging/BUILD.gn
index 1d7f14442f2..3eb735f7472 100644
--- a/abseil-cpp/absl/debugging/BUILD.gn
+++ b/abseil-cpp/absl/debugging/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import("//build/config/sanitizers/sanitizers.gni")
+import("//chromium/build/config/sanitizers/sanitizers.gni")
 import("//third_party/abseil-cpp/absl.gni")
 
 absl_source_set("stacktrace") {
